app/main/helpers/frameworks.py

```diff
@@ -104,13 +111,17 @@ def get_reusable_declaration(supplier_id, client, exclude_framework_slugs=None):
      :return: (declaration, framework)
      """
     exclude_framework_slugs = exclude_framework_slugs or []
-    declarations = {i['frameworkSlug']: i for i in client.find_supplier_declarations(supplier_id)['frameworkInterest']}
+    supplier_frameworks = client.find_supplier_declarations(supplier_id)['frameworkInterest']
+    supplier_frameworks_with_completed_declarations = filter(
```

Comment:
Fortunately `get_supplier_framework_info` returns the declaration with the framework info so I swapped out `get_supplier_declaration` for `get_supplier_framework_info`